hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/HashJoinP.java

```diff
@@ -139,6 +139,10 @@ public HashJoinP(
         });
     }
 
+    private Object matchedValue(Object joinedObject) {
+        return joinedObject == NONE ? null : joinedObject;
```

Comment:
Can't we use `null` directly instead of `NONE`? I don't see where a distinction between those is needed.